lib/connectors/mongodb/connector.rb

```diff
@@ -72,6 +74,8 @@ def do_health_check
       end
 
       def with_client
+        raise "Invalid value for 'Direct connection' : #{@direct_connection}." unless %w[true false].include?(@direct_connection.to_s.strip.downcase)
```

Comment:
I didn't validate in `initialize`. In that case, the job runner will fail early and sync job won't be claimed, therefore it won't be surfaced to Kibana.